plugins/catalog-backend/src/ingestion/processors/util/org.ts

```diff
@@ -100,15 +55,24 @@ export function buildMemberOf(groups: GroupEntity[], users: UserEntity[]) {
   const groupsByName = new Map(groups.map(g => [g.metadata.name, g]));
 
   users.forEach(user => {
```

Comment:
The documentation states that transitive membership should NOT be specified - so it would seem that this code should not be needed. But I chose to keep this behaviour for now (same functionality, but using the new ancestor-less data) instead of changing that concern as well.

Some background:

- We want the emitted _relations_ of user membership in groups to be transitive in the end.
- But we don't want people to HAVE to fill in all of the transitive memberOf values _in the spec_.
- But the rub is, we need to have a global view of all groups, to be able to know the org hierarchy - because groups do not either reference their ancestors and descendants after this PR is merged.
- So it won't be possible to write a post-processor that looks at one user at a time, and emits its transitive membership relations.
- That means that the _reader processor_ probably will be the one that emits those relations, because it's the one that knows of all of the groups at once before emitting them.

Reply:
Yeah even though I first wanted to skip the transitive ones, your points are exactly the reasons why I ended up with that behavior in the end.

Reply:
Mm. So we'll be in a bit of an intermediate state where we have to figure out where to go with this.

One solution could of course be to continue to have org reader processors call this helper method to actually put the full list in the spec.

- Benefit: To those processors, there's no harm done, it's just one call away to fill in the entire model.
- Benefit: We can make a core post-processor that automatically emits relations from this filled-in field, instead of making all reader processors emit this relation.
- Benefit: We just have to tweak the docs slightly.
- Drawback: There won't be a distinction between direct and indirect memberships, so a consumer that might be interested in seeing the direct ones and render them won't be able to. But this isn't a usage that we're targeting either way. If we want to do that, we may have to rethink the model a bit.
- Drawback: Other use cases like reading hand-written files on disk will not get the transitives filled in, which means that transitive ownership won't work for those cases. But maybe they are rare particularly in production.